src/simplnx/Utilities/ClusteringUtilities.cpp

```diff
@@ -0,0 +1,44 @@
+#include "ClusteringUtilities.hpp"
+
+#include "simplnx/Utilities/DataStoreUtilities.hpp"
+
+#include <random>
+
+namespace nx::core::ClusterUtilities
+{
+void RandomizeFeatureIds(Int32AbstractDataStore& featureIdsStore, usize totalFeatures)
+{
+  const usize rangeMin = 1;
+  const usize rangeMax = totalFeatures - 1;
+  auto gen = std::mt19937_64(std::mt19937_64::default_seed);
+  std::uniform_real_distribution<float64> dist(0, 1);
+
+  IDataStore::ShapeType tupleShape{totalFeatures};
+  IDataStore::ShapeType componentShape{1};
+  std::shared_ptr<AbstractDataStore<int32>> randomIdsPtr = nx::core::DataStoreUtilities::CreateDataStore<int32>(tupleShape, componentShape, IDataAction::Mode::Execute);
+  Int32AbstractDataStore& randomIds = *randomIdsPtr.get();
```

Comment:
The previous implementation used `std::vector`. Since the data isn't exposed outside of this function I think `std::vector` is still probably the best choice.

Reply:
Ah. Can `DataStoreUtilities::CreateDataStore` create out of core arrays?